src/applications/vaos/utils/selectors.js

```diff
@@ -132,28 +132,24 @@ export function getCCEType(state) {
   return typeOfCare?.cceType;
 }
 
-export function getSelectedTypeOfCareFacilities(state) {
-  const data = getFormData(state);
-  const newAppointment = getNewAppointment(state);
-  const typeOfCare = getTypeOfCare(data);
-  return newAppointment.facilities[(typeOfCare?.id)];
-}
-
 export function getParentFacilities(state) {
   return getNewAppointment(state).parentFacilities;
 }
 
-export function getChosenFacilityInfo(state) {
+export function getTypeOfCareFacilities(state) {
   const data = getFormData(state);
   const facilities = getNewAppointment(state).facilities;
   const typeOfCareId = getTypeOfCare(data)?.id;
-  const selectedTypeOfCareFacilities = selectUseFlatFacilityPage(state)
+
+  return selectUseFlatFacilityPage(state)
     ? facilities[`${typeOfCareId}`]
     : facilities[`${typeOfCareId}_${data.vaParent}`];
+}
 
+export function getChosenFacilityInfo(state) {
   return (
-    selectedTypeOfCareFacilities?.find(
-      facility => facility.id === data.vaFacility,
+    getTypeOfCareFacilities(state)?.find(
+      facility => facility.id === getFormData(state).vaFacility,
     ) || null
   );
 }
```

Comment:
Split the selection of type of care facilities into it's own helper function `getTypeOfCareFacilities()` so we can use it elsewhere.